go/mysql/datetime/datetime.go

```diff
@@ -719,6 +719,52 @@ func NewTimeFromStd(t time.Time) Time {
 	}
 }
 
+func NewTimeFromSecondsDecimal(seconds decimal.Decimal) Time {
+	var neg bool
+	if seconds.Cmp(decimal.NewFromInt(0)) < 0 {
+		neg = true
+		seconds = seconds.Mul(decimal.NewFromInt(-1))
+	}
+
+	sec, frac := seconds.QuoRem(decimal.New(1, 0), 0)
+	ns := frac.Mul(decimal.New(1, 9))
+
+	h := sec.Div(decimal.NewFromInt(3600), 0)
+	_, sec = sec.QuoRem(decimal.NewFromInt(3600), 0)
+	min := sec.Div(decimal.NewFromInt(60), 0)
+	_, sec = sec.QuoRem(decimal.NewFromInt(60), 0)
+
+	if h.Cmp(decimal.NewFromInt(839)) >= 0 {
```

Comment:
Shall we extract 838 or 839 into a constant? It's also used in `fn_time.go` as well, so maybe time to do that? I'd probably say I have a tiny preference for checks in the style of `> 838` vs. `>= 839`, but either works then really. 

We can also use it in the next line then.

Reply:
done.